spring-cloud-zookeeper-discovery/src/main/java/org/springframework/cloud/zookeeper/discovery/dependency/DependenciesBasedLoadBalancer.java

```diff
@@ -56,6 +60,7 @@ public Server chooseServer(Object key) {
 		};
 		cacheEntryIfMissing(keyAsString, dependency);
 		log.debug("Will try to retrieve dependency for key [{}]. Current cache contents [{}]", keyAsString, this.ruleCache);
+		updateListOfServers();
```

Comment:
I'm worried that `initWithNiwsConfig` isn't called. It contains a call to `updateListOfServers` among other things.